staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -110,33 +115,48 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 		proxy = config.Proxy
 	}
 
-	// Cache a single transport for these options
-	c.transports[key] = utilnet.SetTransportDefaults(&http.Transport{
+	transport := utilnet.SetTransportDefaults(&http.Transport{
 		Proxy:               proxy,
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     tlsConfig,
 		MaxIdleConnsPerHost: idleConnsPerHost,
 		DialContext:         dial,
 		DisableCompression:  config.DisableCompression,
+		IdleConnTimeout:     idleConnTimeout,
 	})
-	return c.transports[key], nil
+
+	if canCache {
+		// Cache a single transport for these options
+		c.transports[key] = transport
+	} else {
+		/*
+		   verify we don't have in-tree things hot-looping creating clients (add logging to this method and check invocations on all components in e2e/scalability run)
+		   TODO(roycaihw): remove this log before submit
+		*/
+		klog.V(1).Infof(">>> created an uncacheable transport for: %v", config.UserAgent)
```

Comment:
might capture the first line in the stacktrace outside client-go to try to see where this is being constructed from, in case the user-agent isn't helpful